tests/monitoring/monitoring_service/factories.py

```diff
@@ -0,0 +1,77 @@
+import random
+
+from monitoring_service.states import (
+    Channel,
+    HashedBalanceProof,
+    MonitorRequest,
+    OnChainUpdateStatus,
+    UnsignedMonitorRequest,
+)
+from raiden.constants import UINT64_MAX, UINT256_MAX
+from raiden.utils.typing import (
+    BlockNumber,
+    ChainID,
+    ChannelID,
+    Nonce,
+    TokenAmount,
+    TokenNetworkAddress,
+)
+from raiden_contracts.constants import ChannelState
+from raiden_libs.types import TransactionHash
+from raiden_libs.utils import private_key_to_address
+
+DEFAULT_TOKEN_NETWORK_ADDRESS = TokenNetworkAddress(bytes([1] * 20))
+DEFAULT_CHANNEL_IDENTIFIER = ChannelID(3)
+DEFAULT_PRIVATE_KEY1 = "0x" + "1" * 64
+DEFAULT_PRIVATE_KEY2 = "0x" + "2" * 64
+DEFAULT_PARTICIPANT1 = private_key_to_address(DEFAULT_PRIVATE_KEY1)
+DEFAULT_PARTICIPANT2 = private_key_to_address(DEFAULT_PRIVATE_KEY2)
+DEFAULT_REWARD_AMOUNT = TokenAmount(1)
+DEFAULT_SETTLE_TIMEOUT = 100
+
+
+def create_signed_monitor_request(
+    nonce: Nonce = Nonce(5),
+    reward_amount: TokenAmount = DEFAULT_REWARD_AMOUNT,
+    closing_privkey: str = DEFAULT_PRIVATE_KEY1,
+    nonclosing_privkey: str = DEFAULT_PRIVATE_KEY2,
+) -> MonitorRequest:
+    bp = HashedBalanceProof(
+        channel_identifier=DEFAULT_CHANNEL_IDENTIFIER,
+        token_network_address=DEFAULT_TOKEN_NETWORK_ADDRESS,
+        chain_id=ChainID(1),
+        balance_hash="",
+        nonce=nonce,
+        additional_hash="",
+        priv_key=closing_privkey,
+    )
+    monitor_request = UnsignedMonitorRequest.from_balance_proof(
+        bp, reward_amount=reward_amount
+    ).sign(nonclosing_privkey)
+
+    # Some signature correctness checks
+    assert monitor_request.signer == private_key_to_address(closing_privkey)
+    assert monitor_request.non_closing_signer == private_key_to_address(nonclosing_privkey)
+    assert monitor_request.reward_proof_signer == private_key_to_address(nonclosing_privkey)
+
+    return monitor_request
+
+
+def get_random_channel_identifier() -> ChannelID:
+    return ChannelID(random.randint(0, UINT256_MAX))
+
+
+def create_channel(update_status: OnChainUpdateStatus = None) -> Channel:
+    return Channel(
+        token_network_address=DEFAULT_TOKEN_NETWORK_ADDRESS,
+        identifier=DEFAULT_CHANNEL_IDENTIFIER,
+        participant1=DEFAULT_PARTICIPANT1,
+        participant2=DEFAULT_PARTICIPANT2,
+        settle_timeout=random.randint(0, UINT256_MAX),
+        state=random.choice(list(ChannelState)),
+        closing_block=BlockNumber(random.randint(0, UINT256_MAX)),
+        closing_participant=DEFAULT_PARTICIPANT1,
+        closing_tx_hash=TransactionHash("%d" % random.randint(0, UINT64_MAX)),
+        claim_tx_hash=TransactionHash("%d" % random.randint(0, UINT64_MAX)),
```

Comment:
Aren't transaction hashes hex encoded?
(At least until we do https://github.com/raiden-network/raiden-services/issues/364)

Reply:
Yes, the `TransactionHash` is our type from `raiden_libs.types` if that's what you wonder.

Reply:
The we should use `%x` instead of `%d`.